synapse/logging/opentracing.py

```diff
@@ -909,8 +909,9 @@ def tag_args(func: Callable[P, R]) -> Callable[P, R]:
     @wraps(func)
     def _tag_args_inner(*args: P.args, **kwargs: P.kwargs) -> R:
         argspec = inspect.getfullargspec(func)
-        for i, arg in enumerate(argspec.args[1:]):
-            set_tag("ARG_" + arg, str(args[i]))  # type: ignore[index]
+        # We use `[1:]` to skip the `self` object reference
+        for i, arg in enumerate(args[1:]):
+            set_tag("ARG_" + argspec.args[i + 1], str(arg))  # type: ignore[index]
         set_tag("args", str(args[len(argspec.args) :]))  # type: ignore[index]
```

Comment:
What is this one targeting exactly? I'm having a hard time thinking of a scenario? Just extra args passed to the function that weren't in the function signature?

Reply:
> Just extra args passed to the function that weren't in the function signature?

Yes, I think it's designed to handle a function which accepts a variadic number of positional arguments, e.g.
`def foo(first, second, *extras): ...`.
 
The invocation `foo(1, 2, 3, 4)` is perfectly legitimate. But "slots" corresponding to the values 3 and 4 won't show up in `argspec.args`.

```python
>>> def foo(first, second, *extras): ...
... 
>>> import inspect
>>> spec = inspect.getfullargspec(foo)
>>> spec
FullArgSpec(args=['first', 'second'], varargs='extras', varkw=None, defaults=None, kwonlyargs=[], kwonlydefaults=None, annotations={})
>>> spec.args
['first', 'second']
```

Reply:
Thank you for explaining @DMRobertson 😀